presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -403,6 +403,72 @@ public static Slice upper(@SqlType("varchar(x)") Slice slice)
         return toUpperCase(slice);
     }
 
+    private static Slice pad(Slice text, long inputResultSize, Slice fillCharacters, int paddingOffset)
+    {
+        checkCondition(text.length() > 0, INVALID_FUNCTION_ARGUMENT, "Input string must not be empty");
+        checkCondition(inputResultSize > 0, INVALID_FUNCTION_ARGUMENT, "Result size must be greater than 0");
+        checkCondition(fillCharacters.length() > 0, INVALID_FUNCTION_ARGUMENT, "Fill character(s) must not be empty");
+
+        int textLength = countCodePoints(text);
+        int resultSize = Ints.saturatedCast(inputResultSize);
```

Comment:
This should be a checkedCast. It probably won't matter in practice, but, in general, we'd rather fail that produce possibly incorrect results. Also, you should add a `checkCondition` above for `inputResultSize <= Integer.MAX_VALUE` so that Presto raises a proper error if the value is out of range.